### PR TITLE
ftgl: 2.1.3-rc5 -> 2.4.0; cimg: 2.9.6 -> 2.9.7; dialog: 1.3-2210306 -> 1.3-2210324; docutils: 0.16 -> 0.17.1; fme: refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

Repology is haunting me (just kidding, I am haunting myself)

ftgl: 2.1.3-rc5 -> 2.4.0; 
cimg: 2.9.6 -> 2.9.7; 
dialog: 1.3-2210306 -> 1.3-2210324; 
docutils: 0.16 -> 0.17.1; 
fme: refactor

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120625` at 3482685f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>885 packages marked as broken and skipped:</summary>
  <ul>
    <li>adobe-reader</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-openj9-bin-13</li>
    <li>adoptopenjdk-jre-openj9-bin-14</li>
    <li>adoptopenjdk-openj9-bin-13</li>
    <li>adoptopenjdk-openj9-bin-14</li>
    <li>agdaPackages.iowa-stdlib</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>20729 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>CHOWTapeModel</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>MACS2</li>
    <li>MIDIVisualizer</li>
    <li>MMA</li>
    <li>OSCAR</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>183 packages built successfully:</summary>
  <ul>
    <li>asciidoctor</li>
    <li>atk (gnome2.atk ,gnome3.atk)</li>
    <li>cairo (gnome2.cairo)</li>
    <li>cimg</li>
    <li>desktop-file-utils</li>
    <li>dialog</li>
    <li>docutils (python38Packages.docutils)</li>
    <li>fontconfig</li>
    <li>fontforge (fontforge-fonttools ,python38Packages.fontforge)</li>
    <li>freeglut</li>
    <li>ftgl</li>
    <li>gd</li>
    <li>gdk-pixbuf</li>
    <li>gi-docgen</li>
    <li>glib (gnome2.glib ,gnome3.glib)</li>
    <li>gtk-doc (gnome2.gtkdoc)</li>
    <li>gnome2.libIDL</li>
    <li>pango (gnome2.pango)</li>
    <li>gobject-introspection (gnome3.gobject-introspection)</li>
    <li>gsettings-desktop-schemas (gnome3.gsettings-desktop-schemas)</li>
    <li>gts</li>
    <li>harfbuzz</li>
    <li>jpylyzer (python38Packages.jpylyzer)</li>
    <li>json-glib</li>
    <li>lasem</li>
    <li>libass</li>
    <li>libdrm</li>
    <li>libnotify</li>
    <li>libpsl</li>
    <li>libqrtr-glib</li>
    <li>perl532Packages.X11Protocol</li>
    <li>python38Packages.Babel</li>
    <li>python38Packages.Mako</li>
    <li>python38Packages.alabaster</li>
    <li>python38Packages.apipkg</li>
    <li>python38Packages.astor</li>
    <li>python38Packages.async_generator</li>
    <li>python38Packages.automat</li>
    <li>python38Packages.backcall</li>
    <li>python38Packages.backports-entry-points-selectable</li>
    <li>python38Packages.betamax</li>
    <li>python38Packages.blinker</li>
    <li>python38Packages.brotli</li>
    <li>python38Packages.chardet</li>
    <li>python38Packages.cleo</li>
    <li>python38Packages.click</li>
    <li>python38Packages.clikit</li>
    <li>python38Packages.cloudpickle</li>
    <li>python38Packages.convertdate</li>
    <li>python38Packages.ddt</li>
    <li>python38Packages.django (python38Packages.django_2)</li>
    <li>python38Packages.entrypoints</li>
    <li>python38Packages.et_xmlfile</li>
    <li>python38Packages.execnet</li>
    <li>python38Packages.flake8</li>
    <li>python38Packages.flaky</li>
    <li>python38Packages.flit-core</li>
    <li>python38Packages.freezegun</li>
    <li>python38Packages.gunicorn</li>
    <li>python38Packages.hijri-converter</li>
    <li>python38Packages.html5lib</li>
    <li>python38Packages.hyperlink</li>
    <li>python38Packages.hypothesis</li>
    <li>python38Packages.idna</li>
    <li>python38Packages.intreehooks</li>
    <li>python38Packages.iso8601</li>
    <li>python38Packages.jdcal</li>
    <li>python38Packages.jinja2</li>
    <li>python38Packages.jupyterlab-pygments</li>
    <li>python38Packages.libevdev</li>
    <li>python38Packages.m2r</li>
    <li>python38Packages.mccabe</li>
    <li>python38Packages.multidict</li>
    <li>python38Packages.munkres</li>
    <li>python38Packages.openpyxl</li>
    <li>python38Packages.pastel</li>
    <li>python38Packages.pathpy</li>
    <li>python38Packages.pep517</li>
    <li>python38Packages.pickleshare</li>
    <li>python38Packages.psutil</li>
    <li>python38Packages.py-cpuinfo</li>
    <li>python38Packages.pyasn1-modules</li>
    <li>python38Packages.pycairo</li>
    <li>python38Packages.pydocstyle</li>
    <li>python38Packages.pygments</li>
    <li>python38Packages.pygobject3</li>
    <li>python38Packages.pylama</li>
    <li>python38Packages.pymeeus</li>
    <li>python38Packages.pyroma</li>
    <li>python38Packages.pytest (python38Packages.pytest_6)</li>
    <li>python38Packages.pytest-asyncio</li>
    <li>python38Packages.pytest-benchmark</li>
    <li>python38Packages.pytestcov (python38Packages.pytest-cov)</li>
    <li>python38Packages.pytest-expect</li>
    <li>python38Packages.pytest-flake8</li>
    <li>python38Packages.pytest-forked</li>
    <li>python38Packages.pytest-freezegun</li>
    <li>python38Packages.pytest-mock</li>
    <li>python38Packages.pytestrunner (python38Packages.pytest-runner)</li>
    <li>python38Packages.pytest-subtests</li>
    <li>python38Packages.pytest-timeout</li>
    <li>python38Packages.pytest-xdist (python38Packages.pytest_xdist)</li>
    <li>python38Packages.pytestCheckHook</li>
    <li>python38Packages.pytoml</li>
    <li>python38Packages.requests</li>
    <li>python38Packages.responses</li>
    <li>python38Packages.send2trash</li>
    <li>python38Packages.simplejson</li>
    <li>python38Packages.smartypants</li>
    <li>sphinx (python38Packages.sphinx)</li>
    <li>python38Packages.sqlparse</li>
    <li>python38Packages.text-unidecode</li>
    <li>python38Packages.typogrify</li>
    <li>python38Packages.urllib3</li>
    <li>python38Packages.webencodings</li>
    <li>python38Packages.whoosh</li>
    <li>python38Packages.yarl</li>
    <li>python39Packages.Babel</li>
    <li>python39Packages.alabaster</li>
    <li>python39Packages.apipkg</li>
    <li>python39Packages.astor</li>
    <li>python39Packages.async_generator</li>
    <li>python39Packages.automat</li>
    <li>python39Packages.backcall</li>
    <li>python39Packages.backports-entry-points-selectable</li>
    <li>python39Packages.blinker</li>
    <li>python39Packages.click</li>
    <li>python39Packages.cloudpickle</li>
    <li>python39Packages.ddt</li>
    <li>python39Packages.docutils</li>
    <li>python39Packages.entrypoints</li>
    <li>python39Packages.et_xmlfile</li>
    <li>python39Packages.execnet</li>
    <li>python39Packages.flake8</li>
    <li>python39Packages.flaky</li>
    <li>python39Packages.flit-core</li>
    <li>python39Packages.freezegun</li>
    <li>python39Packages.gunicorn</li>
    <li>python39Packages.hijri-converter</li>
    <li>python39Packages.hyperlink</li>
    <li>python39Packages.idna</li>
    <li>python39Packages.intreehooks</li>
    <li>python39Packages.iso8601</li>
    <li>python39Packages.jdcal</li>
    <li>python39Packages.jinja2</li>
    <li>python39Packages.jupyterlab-pygments</li>
    <li>python39Packages.m2r</li>
    <li>python39Packages.mccabe</li>
    <li>python39Packages.multidict</li>
    <li>python39Packages.openpyxl</li>
    <li>python39Packages.pastel</li>
    <li>python39Packages.pathpy</li>
    <li>python39Packages.pep517</li>
    <li>python39Packages.pickleshare</li>
    <li>python39Packages.psutil</li>
    <li>python39Packages.pyasn1-modules</li>
    <li>python39Packages.pydocstyle</li>
    <li>python39Packages.pygments</li>
    <li>python39Packages.pylama</li>
    <li>python39Packages.pymeeus</li>
    <li>python39Packages.pytest (python39Packages.pytest_6)</li>
    <li>python39Packages.pytest-asyncio</li>
    <li>python39Packages.pytestcov (python39Packages.pytest-cov)</li>
    <li>python39Packages.pytest-expect</li>
    <li>python39Packages.pytest-flake8</li>
    <li>python39Packages.pytest-forked</li>
    <li>python39Packages.pytest-freezegun</li>
    <li>python39Packages.pytest-mock</li>
    <li>python39Packages.pytestrunner (python39Packages.pytest-runner)</li>
    <li>python39Packages.pytest-subtests</li>
    <li>python39Packages.pytestCheckHook</li>
    <li>python39Packages.pytoml</li>
    <li>python39Packages.send2trash</li>
    <li>python39Packages.simplejson</li>
    <li>python39Packages.sqlparse</li>
    <li>python39Packages.text-unidecode</li>
    <li>python39Packages.tzdata</li>
    <li>python39Packages.webencodings</li>
    <li>python39Packages.yarl</li>
    <li>shared-mime-info</li>
    <li>volk</li>
    <li>xlibsWrapper (xorg.xlibsWrapper)</li>
    <li>xorg.libXft</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

bc is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/misc/fme/default.nix:29:3:
```
   |
29 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `isPy38`.
Near pkgs/development/python-modules/docutils/default.nix:5:3:
```
  |
5 | , isPy38
  |   ^
```
</li>
  </ul>
</details>

**Comment by zowoq** [comment]
Reverted the docutils bump in https://github.com/NixOS/nixpkgs/commit/fbf8ee0a701b3e28ae25c5263e8af2938621ca61 as it causes a mass rebuild.

**Comment by AndersonTorres** [comment]
Seriously? Oh man.

**Comment by SuperSandro2000** [comment]
> Seriously? 

Ofborg labeled it like this.

File: pkgs/applications/misc/fme/default.nix [hunk 1]
```diff
@@ -1,33 +1,56 @@
...
+, libsigcxx
+, pkg-config
+}:
 
 stdenv.mkDerivation rec {
-
   pname = "fme";
   version = "1.1.3";
 
   src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/fme/default.nix:20]: Please use fetchFromGitHub.

File: pkgs/development/libraries/cimg/default.nix [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
     cp -dr --no-preserve=ownership examples/* $doc/share/doc/cimg/examples/
     cp -dr --no-preserve=ownership plugins/* $out/include/CImg/plugins/
     cp README.txt $doc/share/doc/cimg/
+    runHook postInstall
   '';
 
   outputs = [ "out" "doc" ];
 
   meta = with lib; {
+    homepage = "http://cimg.eu/";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cimg/default.nix:30]: Redirects to https://cimg.eu/.

File: pkgs/development/python-modules/docutils/default.nix [hunk 3]
```diff
@@ -30,8 +29,8 @@ buildPythonPackage rec {
   '';
 
   meta = with lib; {
-    description = "Python Documentation Utilities";
     homepage = "http://docutils.sourceforge.net/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/docutils/default.nix:32]: Redirects to https://docutils.sourceforge.io/

**Comment by AndersonTorres** [comment]
> > Seriously?
> 
> Ofborg labeled it like this.

It was a bit surprising.

**Pull request merged**